concordium-node/src/network/buckets.rs

```diff
@@ -105,16 +114,23 @@ impl Buckets {
     }
 
     /// Removes the bucket nodes older than then specified amount of time.
-    pub fn clean_buckets(&mut self, timeout_bucket_entry_period: u64) {
+    pub fn clean_buckets(
+        &mut self,
+        timeout_bucket_entry_period: u64,
+        bucket_size_gauge: &IntGaugeVec,
+    ) {
         let clean_before = get_current_stamp() - timeout_bucket_entry_period;
         self.buckets[0].retain(|entry| entry.last_seen >= clean_before);
+        let new_bucket_size = self.buckets[0].len();
+        bucket_size_gauge.with_label_values(&["0"]).set(new_bucket_size.try_into().unwrap());
```

Comment:
I think in this case it would be better to cast the size, since we don't want to panic just because of the metrics.
I know it is a bit nitpicking because this can't come into the state where it would panic, but still think it is important to consider and save the reader from having to do the reasoning.
So, something like this:
```suggestion
        bucket_size_gauge.with_label_values(&["0"]).set(new_bucket_size as i64);
```